### PR TITLE
Add MySQLResources and reinstate HealthCheckerTest

### PR DESCRIPTION
**Pull request by ryanemerson**

### CODE REVIEW

File: src/test/java/io/gingersnapproject/MySQLResources.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+public class MySQLResources implements QuarkusTestResourceLifecycleManager {
+
+   private GenericContainer<?> cacheManager;
+   private MySQLContainer<?> db;
+
+   @Override
+   public Map<String, String> start() {
+      cacheManager = new GenericContainer<>("quay.io/gingersnap/cache-manager")
+            .withNetworkAliases("infinispan-" + Base58.randomString(6))
+            .withExposedPorts(8080, 11222)
```
Comment (jabolina) [inline src/test/java/io/gingersnapproject/MySQLResources.java:22]: Maybe we'll have to map 8081 to 8080, so the db-syncer starts using 8080

Comment (ryanemerson) [reply src/test/java/io/gingersnapproject/MySQLResources.java:22]: I don't think that's necessary as TestContainers already exposes this container port differently to what we expose. If you look in the logs you will see something like:

```
2022-12-09 09:28:33,909 INFO  [org.tes.con.wai.str.HttpWaitStrategy] (pool-3-thread-1) /unruffled_swanson: Waiting for 60 seconds for URL: http://localhost:49157/q/health (where port 49157 maps to container port 8080)
```

and when configuring the db-syncer, we use the container mapping for the actual port value:

```java
"gingersnap.cache.uri", String.format("hotrod://%s:%d", cacheManager.getHost(), cacheManager.getMappedPort(11222)),
```

**Review by jabolina — APPROVED** [review]
Looks really nice :smiley:. Should we merge or you're adding more stuff?

**Comment by jabolina** [comment]
Thanks, @ryanemerson.

**Pull request merged**